src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -194,6 +194,47 @@ internal PathInfo SetLocation(string path)
         /// If the <paramref name="path"/> could not be resolved.
         /// </exception>
         internal PathInfo SetLocation(string path, CmdletProviderContext context)
+        {
+            return SetLocation(path, context, literalPath: false);
+        }
+
+        /// <summary>
+        /// Changes the current working directory to the path specified
+        /// </summary>
+        /// <param name="path">
+        /// The path of the new current working directory
+        /// </param>
+        /// <param name="context">
+        /// The context the provider uses when performing the operation.
+        /// </param>
+        /// <param name="literalPath">
+        /// Indicate if the path is a literal path.
+        /// </param>
+        /// <returns>
+        /// The PathInfo object representing the path of the location
+        /// that was set.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// If <paramref name="path"/> is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// If <paramref name="path"/> does not exist, is not a container, or
+        /// resolved to multiple containers.
+        /// </exception>
+        /// <exception cref="ProviderNotFoundException">
+        /// If <paramref name="path"/> refers to a provider that does not exist.
+        /// </exception>
+        /// <exception cref="DriveNotFoundException">
+        /// If <paramref name="path"/> refers to a drive that does not exist.
+        /// </exception>
+        /// <exception cref="ProviderInvocationException">
+        /// If the provider associated with <paramref name="path"/> threw an
+        /// exception.
+        /// </exception>
+        /// <exception cref="ItemNotFoundException">
+        /// If the <paramref name="path"/> could not be resolved.
+        /// </exception>
+        internal PathInfo SetLocation(string path, CmdletProviderContext context, bool literalPath)
```

Comment:
Should it be private now?

Reply:
No, it's needed to be called from PathInterfaces